src/core/service.c

```diff
@@ -3879,6 +3881,14 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                 default:
                                         assert_not_reached();
                                 }
+                        } else if (s->exit_type == SERVICE_EXIT_CGROUP) {
```

Comment:
Please merge the `s->state` if block below with this one. Also, according to our [coding style](https://systemd.io/CODING_STYLE/), please remove the unnecessary `{}` around the if block.

Reply:
I've fixed it. Thank you for your review.